lib-persistence-manager/persistencemgr/redis.go

```diff
@@ -177,8 +177,12 @@ func GetDBConnection(dbFlag DbType) (*ConnPool, *errors.Error) {
 	case InMemory:
 		// In this case this function return in-memory db connection pool
 		if inMemDBConnPool == nil || inMemDBConnPool.ReadPool == nil {
+			log.Info("GetDBConnection : inMemDBConnPool OR inMemDBConnPool.ReadPool is nil")
 			config := getInMemoryDBConfig()
 			inMemDBConnPool, err = config.Connection()
+			if err != nil {
+				log.Error("error while trying to get Inmemory Readpool connection : " + err.Error())
+			}
```

Comment:
shouldn't we return here?

Reply:
There was something Hemantha had said about this. Dont remember but dont change right now